kikit/fab/neodenyy1.py

```diff
@@ -0,0 +1,164 @@
+import click
+from pcbnewTransition import pcbnew
+import csv
+import os
+import sys
+import shutil
+import re
+from pathlib import Path
+from kikit.fab.common import *
+from kikit.common import *
+from kikit.export import gerberImpl
+mm_ius = 1000000.0
+footprintRe = [
```

Comment:
Wouldn't a dictionary better express the intention?

Reply:
Also, it seems to be a constant. So I suggest naming it uppercase.

Reply:
Name it uppercase is done, this regex need to loop for it, so use tuple is more easy.

Reply:
I still suggest changing it to dictionary as it better expresses the intention and you can make the code more readable. Instead of:

```.py
for f in FOOTPRIINTREGEX:
        matchedFootprint = f[0].match(footprint)
        if matchedFootprint != None:
            return f[1].format(matchedFootprint.groups()[0])
```

you can write:
```.py
for pattern, replacement in FOOTPRIINTREGEX.items():
        matchedFootprint = pattern.match(footprint)
        if matchedFootprint != None:
            return replacement.format(matchedFootprint.groups()[0])
```

Instead of non-obvious names `f[0]` and `f[1]` you immediately get the information:
- `FOOTPRIINTREGEX` contains mappings (there is the direction from what to what) and the patterns are unique
- you get clear idea what is a pattern and what is a replacement.

Reply:
Resolved!